api/envoy/api/v2/listener/listener.proto

```diff
@@ -108,14 +110,12 @@ message FilterChainMatch {
   // connection is contained in at least one of the specified subnets. If the
   // parameter is not specified or the list is empty, the source IP address is
   // ignored.
-  // [#not-implemented-hide:]
   repeated core.CidrRange source_prefix_ranges = 6;
 
   // The criteria is satisfied if the source port of the downstream connection
   // is contained in at least one of the specified ports. If the parameter is
   // not specified, the source port is ignored.
-  // [#not-implemented-hide:]
-  repeated google.protobuf.UInt32Value source_ports = 7;
+  repeated uint32 source_ports = 7 [(validate.rules).repeated .items.uint32 = {gte: 1, lte: 65535}];
```

Comment:
Technically no, but this field was never used or implemented, so I think it's fine to change?

Reply:
It looks that [go-control-plane](https://github.com/envoyproxy/go-control-plane) pulls those unimplemented fields anyway, and it already has [`source_ports`](https://github.com/envoyproxy/go-control-plane/blob/e90fc65c48c6e270d72ab798e0d7c441fcc0a807/envoy/api/v2/listener/listener.pb.go#L291) with type `[]*types.UInt32Value`, whereas this changes its type to `[]uint32`, which would break things (I think?).

cc @kyessenov

Reply:
It would break things only if someone was actually using that field, though I can't imagine they would be. I can change it back if you want, just let me know.

Reply:
Prefer not to do any more language breaking API changes in v2. Maybe tag with a `TODO` for v3, we can do these cleanups there.

Reply:
Will revert back.

Reply:
I find it confusing to have experimental fields in stable protos, and I wouldn't discount that someone might be already using those fields (hopefully not in JSONs/YAMLs). It is more appropriate to fix those experimental fields in v3, but I understand the cost of branching the entire parent API version.

Reply:
Actually, since this was `not-implemented-hide`, it's fair to break, even under the new stable API policy.

Reply:
@kyessenov see https://github.com/envoyproxy/envoy/issues/6271#issuecomment-487716345, we are heading towards major versions.

Reply:
OK, given that I can't get validations to work properly w/ the WKT, I'm going to go ahead and merge as-is.